cli/src/torii/routing.rs

```diff
@@ -116,20 +120,28 @@ pub(crate) async fn handle_queries<W: WorldTrait>(
     pagination: Pagination,
     request: VerifiedQueryRequest,
 ) -> Result<Scale<VersionedPaginatedQueryResult>> {
-    let valid_request = request.validate(&wsv, &query_validator)?;
+    let (valid_request, filter) = request.validate(&wsv, &query_validator)?;
     let original_result = valid_request.execute(&wsv)?;
     let total: u64 = original_result
         .len()
         .try_into()
         .map_err(|e: TryFromIntError| QueryError::Conversion(e.to_string()))?;
-    let result = QueryResult(if let Value::Vec(value) = original_result {
-        Value::Vec(value.into_iter().paginate(pagination).collect())
+
+    let filtered_result = QueryResult(if let Value::Vec(value) = original_result {
+        Value::Vec(
+            value
+                .into_iter()
+                .filter(|val| filter.applies(val))
+                .paginate(pagination)
+                .collect(),
+        )
```

Comment:
why is only `Value::Vec` filtered?

Reply:
Added explanation to `predicate.rs`. Basically, it's ambiguous and not necessary.